### ISSUE TITLE
if the built in on screen controls are visible when the fragment or activity is closed, a leak occurs

### ISSUE THREAD

**Issue by spyhunter99** [osmdroid/osmdroid#328]
```
06-04 16:10:44.899 11013-11013/org.osmdroid E/WindowManager: android.view.WindowLeaked: Activity org.osmdroid.StarterMapActivity has leaked window android.widget.ZoomButtonsController$Container{1ae955c4 V.E..... ......I. 0,0-1080,146} that was originally added here
                                                                 at android.view.ViewRootImpl.<init>(ViewRootImpl.java:500)
                                                                 at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:287)
                                                                 at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:87)
                                                                 at android.widget.ZoomButtonsController.setVisible(ZoomButtonsController.java:370)
                                                                 at org.osmdroid.views.MapView$MapViewGestureDetectorListener.onDown(MapView.java:1186)
                                                                 at android.view.GestureDetector.onTouchEvent(GestureDetector.java:559)
                                                                 at org.osmdroid.views.MapView.dispatchTouchEvent(MapView.java:827)
                                                                 at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2687)
                                                                 at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2304)
                                                                 at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2687)
                                                                 at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2304)
                                                                 at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2687)
                                                                 at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2304)
                                                                 at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2687)
                                                                 at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2304)
                                                                 at android.view.ViewGroup.dispatchTransformedTouchEvent(ViewGroup.java:2687)
                                                                 at android.view.ViewGroup.dispatchTouchEvent(ViewGroup.java:2304)
                                                                 at com.android.internal.policy.impl.PhoneWindow$DecorView.superDispatchTouchEvent(PhoneWindow.java:2739)
                                                                 at com.android.internal.policy.impl.PhoneWindow.superDispatchTouchEvent(PhoneWindow.java:1832)
                                                                 at android.app.Activity.dispatchTouchEvent(Activity.java:2979)
                                                                 at com.android.internal.policy.impl.PhoneWindow$DecorView.dispatchTouchEvent(PhoneWindow.java:2700)
                                                                 at android.view.View.dispatchPointerEvent(View.java:9354)
                                                                 at android.view.ViewRootImpl$ViewPostImeInputStage.processPointerEvent(ViewRootImpl.java:4974)
                                                                 at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4735)
                                                                 at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4267)
                                                                 at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4320)
                                                                 at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4286)
                                                                 at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4412)
                                                                 at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4294)
                                                                 at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:4469)
                                                                 at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4267)
                                                                 at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:4320)
                                                                 at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:4286)
                                                                 at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:4294)
                                                                 at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:4267)
                                                                 at android.view.ViewRootImpl.deliverInputEvent(ViewRootImpl.java:6864)
                                                                 at android.view.ViewRootImpl.doProcessInputEvents(ViewRootImpl.java:6745)
                                                                 at android.view.ViewRootImpl.enqueueInputEvent(ViewRootImpl.java:6716)
                                                                 at android.view.ViewRootImpl$WindowInputEventReceiver.onInputEvent(ViewRootImpl.java:6954)
                                                                 at android.view.InputEventReceiver.dispatchInputEvent(InputEventReceiver.java:185)
                                                                 at android.os.MessageQueue.nativePollOnce(Native Method)
                                                                 at android.os.MessageQueue.next(MessageQueue.java:143)
                                                                 at android.os.Looper.loop(Looper.java:130)
                                                                 at android.app.ActivityThread.main(ActivityThread.java:6840)
                                                                 at java.lang.reflect.Method.invoke(Native Method)
                                                                 at java.lang.reflect.Method.invoke(Method.java:372)
                                                                 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
                                                                 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199)

```

**Comment by spyhunter99** [comment]
use the sample app, move the map which is will display the built in on screen zoom controls, then hit the back button and watch log cat.

**Comment by spyhunter99** [comment]
Looks like the only fix is to set use built in zoom controls to false when leaving the activity or fragment.

**Comment by spyhunter99** [comment]
this is an android issue. i don't think we can solve it, closing for now

**Comment by spyhunter99** [comment]
resolved with #736

**Comment by monsieurtanuki** [comment]
Fixed by #1190. Feel free to reopen.
